Makefile

```diff
@@ -414,12 +414,29 @@ enterprise-code-checkin-test:
 		false; \
 	fi
 
-test-pfs:
+test-pfs-server:
+	@# Check that at least 2048 open file descriptors are allowed (smallest number known to work)
+	if [ $$(ulimit -n) -lt 2048 ]; then \
+	  echo "Number of open file descriptors is limited to $$(ulimit -n), which may be too low"; \
+	  echo "Try running 'ulimit -n 2048' to increase this limit"; \
+	  exit 1; \
+	fi
+	@# If etcd is not available, start it in a docker container
+	if ! ETCDCTL_API=3 etcdctl --endpoints=127.0.0.1:2379 get "testkey"; then \
+	  docker run \
+	    --publish 32379:2379 \
+	    quay.io/coreos/etcd:v3.3.5 \
```

Comment:
It would probably make sense to store this in a variable so that upgrading etcd versions will be a little easier.

Reply:
Done